src/Components/Components/src/Rendering/HtmlRenderer.cs

```diff
@@ -47,20 +47,21 @@ protected override Task UpdateDisplayAsync(in RenderBatch renderBatch)
         /// <param name="componentType">The type of the <see cref="IComponent"/>.</param>
         /// <param name="initialParameters">A <see cref="ParameterCollection"/> with the initial parameters to render the component.</param>
         /// <returns>A <see cref="Task"/> that on completion returns a sequence of <see cref="string"/> fragments that represent the HTML text of the component.</returns>
-        public async Task<IEnumerable<string>> RenderComponentAsync(Type componentType, ParameterCollection initialParameters)
+        public async Task<ComponentRenderedText> RenderComponentAsync(Type componentType, ParameterCollection initialParameters)
         {
-            var frames = await CreateInitialRenderAsync(componentType, initialParameters);
+            var (componentId, frames) = await CreateInitialRenderAsync(componentType, initialParameters);
 
             if (frames.Count == 0)
             {
-                return Array.Empty<string>();
+                // The component didn't produce any output.
+                return new ComponentRenderedText(componentId, Array.Empty<string>());
```

Comment:
Is there a reason why this needs to be a special case? would this *just work* if you let it call through to `RenderFrames`?

Reply:
Updated